### PR TITLE
Fix resource schema validation issues

### PR DESCRIPTION
**Pull request by jormello**
**Description of changes:**

Both the `state machine` and `activity` resources were generating warnings when running `cfn validate` to check the validity of their resource schemas. 

The warnings noted that an entry for `tagging` [property](https://github.com/aws-cloudformation/cloudformation-resource-schema/blob/master/src/main/resources/schema/provider.definition.schema.v1.json#L121-L154) is required to define what tagging behaviour is supported. 

There was also a warning that the [insertionOrder property](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-schema.html#resource-type-schema-properties) is needed for any property of type `array`.  The `insertionOrder` property is `set to true to specify that the order in which array items are specified must be honored, and that changing the order of the array will indicate a change in the property.`

----

**Testing:**
- `mvn package` passing for both resources
- `cfn validate` passing with no warnings for both resources

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: statemachine/aws-stepfunctions-statemachine.json [hunk 1]
```diff
@@ -170,13 +172,21 @@
...
+  },
   "additionalProperties": false,
   "readOnlyProperties": [
     "/properties/Arn",
     "/properties/Name"
   ],
   "createOnlyProperties": [
-    "/properties/StateMachineName"
+    "/properties/StateMachineName",
+    "/properties/StateMachineType"
```
Comment (wong-a) [inline statemachine/aws-stepfunctions-statemachine.json:189]: Please separate unrelated changes. Changing the update behaviour for this property deserves a separate PR/commit. This wasn't a validation issue as the PR title says, is it?

**Review by wong-a — CHANGES_REQUESTED** [review]
The changes themselves LGTM, comments related to PR hygiene. 

Please call out the behaviour change to `StateMachineType` explicitly (update requires replacement) and link the related GitHub issue. Also, is there a way to test this?

File: .gitignore [hunk 2]
```diff
@@ -17,8 +17,5 @@ wrapper
 **/*.zip
 **/*.gz
 
-# auto-generated docs
-**/docs
-
```
[the remark this answers is not in the document]
Comment (jormello) [reply .gitignore]: Fair callout. New reverted those changes in this PR and have opened a new PR #31 to capture the changes for `StateMachineType`.

File: statemachine/aws-stepfunctions-statemachine.json [hunk 3]
```diff
@@ -170,13 +172,21 @@
   "required": [
     "RoleArn"
   ],
+  "tagging": {
+    "taggable": true,
+    "tagOnCreate": true,
+    "tagUpdatable": true,
+    "cloudFormationSystemTags": true,
+    "tagProperty": "/properties/Tags"
+  },
```
[the remark this answers is not in the document]
Comment (jormello) [reply statemachine/aws-stepfunctions-statemachine.json:181]: The insertionOrder parameter is needed, even though we are using the default. Excluding it results in a warning when running `cfn validate`.

[the remark this answers is not in the document]
Comment (jormello) [reply statemachine/aws-stepfunctions-statemachine.json:181]: For `tagging`, we do not need to include any properties beyond `taggable`, but I have included them for clarity.

File: statemachine/aws-stepfunctions-statemachine.json (hunk 1 above)
Comment (jormello) [reply statemachine/aws-stepfunctions-statemachine.json]: As noted above, separate PR #31 opened for `StateMachineType` changes, and #30 for documentation changes.

**Review by wong-a — APPROVED** [review]
